addons/Lookup/Lookup.lua

```diff
@@ -0,0 +1,190 @@
+	WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+	DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY
+	DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+	(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+	LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+	ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+	(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+_addon.name = 'Lookup'
+_addon.author = 'Karuberu'
+_addon.version = '1.0'
+_addon.language = 'english'
+_addon.command = 'lookup'
+
+config = require('config')
+res = require('resources')
+
+settings = nil
+ids = nil
+last_item = nil
+
+function initialize()
+	load_settings()
+	initialize_ids()
+end
+
+function load_settings()
+	settings = config.load({
+		default = 'ffxiclopedia';
+		url = {
+			ffxiclopedia = 'http://ffxiclopedia.wikia.com/wiki/Special:Search?go=Go&search=%s';
+			bgwiki = 'https://www.bg-wiki.com/bg/Special:Search?go=Go&search=%s';
+			ffxidb = {
+				item = 'http://www.ffxidb.com/items/%s';
+				zone = 'http://www.ffxidb.com/zones/%s';
+				search = 'http://www.ffxidb.com/search?q=%s';
+			};
+			ffxiah = {
+				item = 'http://www.ffxiah.com/item/%s';
+				search = 'http://www.ffxiah.com/search/item?q=%s';
+				player = 'https://www.ffxiah.com/search/player?name=%s';
+			};
+			google = 'https://www.google.com/search?q=%s';
+		};
+	})
+end
+
+function initialize_ids()
+	ids = {
+		items = {};
+		zones = {};
+	}
+	for item in res.items:it() do
+	    ids.items[item.name] = item.id
+	    ids.items[item.name_log] = item.id
+	end
+	for zone in res.zones:it() do
+			ids.zones[zone.name] = zone.id
+	end
+end
+
+function find_id(name)
+	return {
+		item = ids.items[name];
+		zone = ids.zones[name];
+	}
+end
+
+function get_name(id, list)
+	for key, value in next, list do
+		if value == id then
+			return key
+		end
+	end
+	return nil
+end
+
+function translate(str)
+	return windower.convert_auto_trans(str)
+end
+
+function get_selection(str)
+	local target = str:match('<(.+)>')
+	if target == 'job' then
+		str = windower.ffxi.get_player().main_job_full
+	elseif target == 'subjob' or target == 'sj' then
+		str = windower.ffxi.get_player().sub_job_full
+	elseif target == 'zone' or target == 'area' then
+		if windower.ffxi.get_info().mog_house then
+			str = 'Mog House'
+		else
+			str = get_name(windower.ffxi.get_info().zone, ids.zones)
+		end
+	elseif target == 'lastitem' or target == 'item' then
+		str = get_name(last_item, ids.items)
+	elseif target ~= nil then
+		local mob = windower.ffxi.get_mob_by_target(str:match('<(.+)>'))
+		if mob ~= nil then
+			str = mob.name
+		else
+			str = nil
+		end
+	end
+	return str
+end
+
+function set_default(term)
+	settings.default = term
+	config.save(settings, 'all')
```

Comment:
While it would probably make sense to save for all users by default, I could imagine some people wanting different default sites for different accounts. A crafting mule for example could have a different preference from the main character.